### PR TITLE
Update "backend services for native mobile apps with ASP.NET Core"

### PR DESCRIPTION
**Pull request by HamidMolareza**
- Update and Improve **Controller**:
   - Use **[ApiController]**
      - Remove [FromBody]
      - Remove ModelState.IsValid
   - Use **ControllerBase**
   - Use **CreatedAtAction** for Create action
   - Improve names and Update codes to new style

- Update and Improve **Startup.cs**
   - Remove services.AddMvc() and Add **services.AddControllers ()**
   - Remove app.UseMvc()

- Update and Improve **native-mobile-backend.md** 
   - Add Creating Project for **VSCode**, **Visual Studio for Mac**
   - Update **links**

### CODE REVIEW

File: aspnetcore/mobile/native-mobile-backend.md [hunk 1]
```diff
@@ -46,11 +47,49 @@ public static string RestUrl = "http://192.168.1.207:5000/api/todoitems/{0}";
 
 ## Creating the ASP.NET Core Project
 
-Create a new ASP.NET Core Web Application in Visual Studio. Choose the Web API template and No Authentication. Name the project *ToDoApi*.
+# [Visual Studio](#tab/visual-studio)
```
Comment (Rick-Anderson) [inline aspnetcore/mobile/native-mobile-backend.md:50]: Delete all duplication from https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api
Send them to https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api for the API project.

This topic should only have unique content.

File: aspnetcore/mobile/native-mobile-backend/sample/ToDoApi/src/ToDoApi/Controllers/ToDoItemsController.cs [hunk 2]
```diff
@@ -1,97 +1,85 @@
...
-    {
+namespace ToDoApi.Controllers {
+    [Route ("api/[controller]")]
+    [ApiController]
+    public class ToDoItemsController : ControllerBase {
         private readonly IToDoRepository _toDoRepository;
 
-        public ToDoItemsController(IToDoRepository toDoRepository)
-        {
+        public ToDoItemsController (IToDoRepository toDoRepository) {
```
[the remark this answers is not in the document]
Comment (Rick-Anderson) [reply aspnetcore/mobile/native-mobile-backend/sample/ToDoApi/src/ToDoApi/Controllers/ToDoItemsController.cs:14]: On second thought, removing all duplication with https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api means this is gone.

**Comment by HamidMolareza** [comment]
@Rick-Anderson, Thanks for the review.

After deleting **the repository**, this article will be very similar to [Create a web API with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-3.1&tabs=visual-studio-code), except a small difference such as:
- Very small difference in **the model**
   - Add Notes Property
   - Maybe change the names
- Very small difference in **controller**
   - Add ErrorCode
   - Delete id from Put action (due to compatibility with [mobile project](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/data-cloud/web-services/rest))
- [Prevent over-posting](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-3.1&tabs=visual-studio-code#prevent-over-posting) is extra

**Suggestions:**
Either don't delete the repository, or if we delete it, we can apply these changes:
- Modify the model in [Create a web API with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-3.1&tabs=visual-studio-code) (Add Notes Property and maybe change the names)(It seems that the article is **independent** and there is no problem to changing it)
- Update [mobile project](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/data-cloud/web-services/rest) to be more compatible and **standard** (consider id for Put action)
- Put the [Prevent over-posting](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-3.1&tabs=visual-studio-code#prevent-over-posting) section in a separate article

**Finally**, tell the audience to use [Create a web API with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-3.1&tabs=visual-studio-code) and make few changes.

Otherwise, it seems that except the mentioned changes (which are very small), the rest of the text is duplicate!

**Comment by Rick-Anderson** [comment]
@HamidMolareza  thanks for doing this. My suggestion:

Tell them to follow the tutorial [Create a web API with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-3.1&tabs=visual-studio-code) ***EXCEPT***

- Very small difference in **model**
   - _Notes_ Property (must be added)
   - Property names (maybe change the names)
- Very small difference in **controller**
   - Error Codes (like 409, for [mobile project](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/data-cloud/web-services/rest))
   - Delete id from Put action (due to compatibility with the mobile project)

So I would give them a list of the differences.

**Comment by Rick-Anderson** [comment]
Sorry I wasn't more clear. We don't want any instructions on creating the Web API controller. The instructions need to come from [Tutorial: Create a web API with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-3.1&tabs=visual-studio)

Say something like, follow [Tutorial: Create a web API with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-3.1&tabs=visual-studio) with the following changes:

* Change the model to ...
* another change

That way we don't need to update this tutorial every version.

**Comment by HamidMolareza** [comment]
Yes, I got it. I updated them to show changes in the article.
I convert this into draft, apply the changes, and report again.
Sorry for the delay.

**Pull request closed without merging**